juju/juju
# Adding machine to update-series.

## Description of change

This is a follow on to PR 7677 to add machine series update to the update-series command.  The series in the machine and unit docs are updated provided that the series is supposed by the unit's charm.  The --force flag may be used if the series is not supported by a unit's charm.  If the machine
has no units, there series information will be still be updated. 

Firing off the changed-config hook will follow on.

## QA steps

1. Deploy two charms, one a subordinate. It's recommended to have 1 series supported by the principal charm not supported by the subordinate.
2. juju update-series ; where the series is supported by both the principal and subordinates. This should succeed.
3. juju update-series ; where the series is supported by the principal and not the subordinate. This should fail.
4. juju update-series --force; where the series is supported by the principal and not the subordinate. This should succeed.
5. juju update-series ; where the series is not supported by the principal. This should fail.
6. juju update-series --force; where the series is not supported by the principal. This should succeed.

At any time, the series to be used by an machine can be found in the output of "juju status <machine_name>"

## Documentation changes

Yes, the online list of commands needs to be updated.

## Bug reference

This is a planned project, however a feature request bug was filed as well:
https://bugs.launchpad.net/juju/+bug/1704135
